pytorch_lightning/trainer/properties.py

```diff
@@ -488,6 +489,14 @@ def train_loop(self) -> FitLoop:
         # FIXME(@awaelchli): the current train_loop should be renamed to fit_loop
         return self.fit_loop
 
+    @property
+    def evaluation_loop(self) -> EvaluationDataLoaderLoop:
+        if self.training and self.sanity_checking:
+            return self.fit_loop.validation_loop
+        elif self.validating:
+            return self.validation_loop
+        return self.test_loop
```

Comment:
elif for this one too? to avoid silent correctness

Reply:
Added a MisConfigurationException.

Reply:
I'd say `RuntimeError` would be a better fit, as this would be an error on our side, not the user's

Reply:
Done !